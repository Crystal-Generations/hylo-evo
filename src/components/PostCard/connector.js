import { connect } from 'react-redux'
import { fetchPost } from 'store/actions/posts'
import { getPostById } from 'store/selectors/getPostById'

export function mapStateToProps (state, props) {
  return {
    post: getPostById(state, props)
  }
}

export const mapDispatchToProps = { fetchPost }

export default connect(mapStateToProps, mapDispatchToProps)
