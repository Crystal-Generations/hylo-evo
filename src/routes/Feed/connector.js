import { connect } from 'react-redux'
import { SAMPLE_FEED_ITEMS } from './sampleData'
import { fetchPosts } from './actions'

function mapStateToProps (state) {
  return {
    feedItems: state.posts
  }
}

export const mapDispatchToProps = { fetchPosts }

export default connect(mapStateToProps, mapDispatchToProps)
