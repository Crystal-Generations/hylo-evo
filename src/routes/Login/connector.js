import { connect } from 'react-redux'
import { checkLogin, login, pickError } from './store'

export default connect(pickError, {checkLogin, login})
