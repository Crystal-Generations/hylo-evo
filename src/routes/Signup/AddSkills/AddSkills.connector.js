import { connect } from 'react-redux'
import { setAutocomplete, getAutocomplete, fetchSkills } from './AddSkills.store'

export function mapStateToProps (state, props) {
  const autocomplete = getAutocomplete(state)
  return {
    autocomplete
  }
}
export const mapDispatchToProps = {
  setAutocomplete,
  fetchSkills
}

export default connect(mapStateToProps, mapDispatchToProps)
