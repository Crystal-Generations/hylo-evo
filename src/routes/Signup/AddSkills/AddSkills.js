import React, { Component } from 'react'
import { get } from 'lodash/fp'
import { isEmpty } from 'lodash'
import SignupModalFooter from '../SignupModalFooter'
import LeftSidebar from '../LeftSidebar'

import '../Signup.scss'

export default class AddSkills extends Component {
  clickHandler = (skill) => {
    this.props.addSkill(skill.name)
  }

  getRemainingSkills = () => {
    const addedSkills = []
    for (let index in this.props.skills) {
      const skill = this.props.skills[index]
      addedSkills.push(skill.name)
    }
    return skills.filter(function (skill) {
      return (addedSkills.indexOf(skill.name) === -1)
    })
  }

  submit = () => {
    this.props.goToNextStep()
  }

  previous = () => {
    this.props.goToPreviousStep()
  }

  componentWillMount = () => {
    const { currentUser } = this.props
    if (get('settings.signupInProgress', currentUser) === 'false') this.props.goBack()
  }

  componentDidMount = () => {
    this.props.fetchMySkills()
  }
  render () {
    return <div styleName='flex-wrapper'>
      <LeftSidebar
        header='Share your unique super powers!'
        body="What skills are you known for? The more skills you add, the more relevant the content. It's like magic."
      />
      <div styleName='right-panel'>
        <span styleName='white-text step-count'>STEP 3/4</span>
        <br />
        <div styleName='center'>
          <input
            styleName='signup-input center-text signup-padding large-input-text gray-bottom-border'
            autoFocus
            onChange={this.handleInputChange}
            placeholder={'How can you help?'}
            readOnly
          />
        </div>
        <div>
          {skills && <div styleName='skills'>
            {this.getRemainingSkills().map((skill, index) =>
              <Pill key={index} skill={skill} handler={() => this.clickHandler(skill)} handlerArg={'name'} />
            )}
          </div>}
        </div>
        <div>
          { !isEmpty(this.props.skills) && <div styleName='your-skills'>
            <div styleName='your-skills-label'><span>Your Skills:</span></div>
            <div styleName='your-skills-list'>
              {this.props.skills.map((skill, index) =>
                <Pill key={index} skill={skill} handler={() => this.clickHandler(skill)} handlerArg={'name'} />
              )}
            </div>
          </div>}
        </div>
        <div>
          <SignupModalFooter previous={this.previous} submit={this.submit} continueText={'Boom, Done'} />
        </div>
      </div>
    </div>
  }
}

export function Pill ({skill, handler, handlerArg}) {
  return <span styleName='skill' onClick={() => handler(skill[handlerArg])}>
    {skill.name}
  </span>
}

const skills = [
  {name: 'Writing'},
  {name: 'Design'},
  {name: 'Project Management'},
  {name: 'Photography'},
  {name: 'Facilitation'},
  {name: 'Media'},
  {name: 'Community Organizing'},
  {name: 'Technology'},
  {name: 'Social Media'},
  {name: 'Event Planning'},
  {name: 'Education'},
  {name: 'Communications'}
]
