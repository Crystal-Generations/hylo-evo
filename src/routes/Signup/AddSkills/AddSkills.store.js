import { get } from 'lodash/fp'

export const SET_AUTOCOMPLETE = `SET_AUTOCOMPLETE`
export const FETCH_SKILLS_AUTOCOMPLETE = `FETCH_SKILLS_AUTOCOMPLETE`

export function setAutocomplete (autocomplete) {
  return {
    type: SET_AUTOCOMPLETE,
    payload: autocomplete
  }
}

export function getAutocomplete (state) {
  return state.AddSkills.autocomplete
}

export function fetchSkills (autocomplete) {
  return {
    type: FETCH_SKILLS_AUTOCOMPLETE,
    graphql: {
      query: `query ($autocomplete: String) {
        skills (first: 10, autocomplete: $autocomplete) {
          items {
            id
            name
          }
        }
      }`,
      variables: {
        autocomplete
      }
    },
    meta: {
      extractModel: 'Skill',
      extractQueryResults: {
        getItems: get('payload.data.skills')
      }
    }
  }
}

// Reducer
const defaultState = {
  skills: []
}

export default function reducer (state = defaultState, action) {
  const { error, type, payload } = action
  if (error) return state

  switch (type) {
    case SET_AUTOCOMPLETE:
      return {
        ...state,
        autocomplete: payload
      }
    default:
      return state
  }
}
