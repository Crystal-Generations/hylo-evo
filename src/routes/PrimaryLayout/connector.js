import { connect } from 'react-redux'
import { SAMPLE_COMMUNITY } from 'routes/Feed/sampleData'

const SAMPLE_USER = {
  id: '1',
  firstName: 'Axolotl',
  lastName: 'Jones',
  avatarUrl: 'https://d3ngex8q79bk55.cloudfront.net/user/13986/avatar/1444260480878_AxolotlPic.png'
}

function mapStateToProps (state) {
  return {
    community: SAMPLE_COMMUNITY,
    currentUser: SAMPLE_USER
  }
}

function mapDispatchToProps (dispatch, props) {
  return {}
}

export default connect(mapStateToProps, mapDispatchToProps)
