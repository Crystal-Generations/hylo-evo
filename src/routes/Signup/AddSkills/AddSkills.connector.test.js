import { mapDispatchToProps, mapStateToProps } from './AddSkills.connector'

describe('AddSkills.connector', () => {
  it('should match latest snapshot for mapDispatchToProps', () => {
    expect(mapDispatchToProps).toMatchSnapshot()
  })
  it('should match latest snapshot for mapStateToProps', () => {
    expect(mapStateToProps).toMatchSnapshot()
  })
})
