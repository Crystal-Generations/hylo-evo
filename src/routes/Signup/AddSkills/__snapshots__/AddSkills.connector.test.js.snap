// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSkills.connector should match latest snapshot for mapDispatchToProps 1`] = `[Function]`;

exports[`AddSkills.connector should match latest snapshot for mapStateToProps 1`] = `[Function]`;
